src/metabase/api/pulse.clj

```diff
@@ -353,7 +354,8 @@
   ;; make sure any email addresses that are specified are allowed before sending the test Pulse.
   (doseq [channel channels]
     (pulse-channel/validate-email-domains channel))
-  (pulse/send-pulse! (assoc body :creator_id api/*current-user-id*))
+  (binding [notification/*send-notification!* notification/send-notification-sync!]
```

Comment:
`pulse/send-pulse!` now calls `notification/*send-notification!*` which is async by default. We don't want to change the behavior here so we need to bind it to the sync function.